src/test/java/gomedic/logic/parser/findcommandparser/FindActivityCommandParserTest.java

```diff
@@ -0,0 +1,51 @@
+package gomedic.logic.parser.findcommandparser;
+
+import static gomedic.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static gomedic.logic.parser.CommandParserTestUtil.assertParseSuccess;
+
+import java.util.Arrays;
+
+import gomedic.logic.commands.findcommand.FindActivityCommand;
+import gomedic.logic.commands.findcommand.FindPatientCommand;
+import gomedic.model.activity.Activity;
+import gomedic.model.person.patient.Patient;
+import gomedic.model.util.ActivityTitleContainsKeywordsPredicate;
+import org.junit.jupiter.api.Test;
+
+import gomedic.commons.core.Messages;
+import gomedic.model.util.NameContainsKeywordsPredicate;
+
+public class FindActivityCommandParserTest{
+
+    private final FindActivityCommandParser parser = new FindActivityCommandParser();
+
+    @Test
+    public void parse_emptyArg_throwsParseException() {
+        assertParseFailure(parser, "     ",
+                String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, FindActivityCommand.MESSAGE_USAGE));
+    }
+
+    @Test
+    public void parse_validArgs_returnsFindDoctorCommand() {
+        // no leading and trailing whitespaces
+        FindActivityCommand expectedFindActivityCommand =
+                new FindActivityCommand(new ActivityTitleContainsKeywordsPredicate<Activity>(Arrays.asList("Alice", "Bob")));
+        assertParseSuccess(parser, "ti/Alice Bob", expectedFindActivityCommand);
+
+        // multiple whitespaces between keywords
+        assertParseSuccess(parser, " ti/ Alice  Bob ", expectedFindActivityCommand);
+
+    }
+    @Test
+    public void parse_invalidArgs_throwsParseException() {
+        // no flag supplied
+        assertParseFailure(parser, " alice bob",
+                String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, FindActivityCommand.MESSAGE_USAGE));
+
+        // multiple flags supplied
+        assertParseFailure(parser, " ti/ alice ti/diabetes ",
```

Comment:
By right, I think if you use the parser given, they would just take the last flag in multiple flags instead of making it fails